internal/controllers/import_controller.go

```diff
@@ -475,3 +493,53 @@ func (r *CAPIImportReconciler) createObject(ctx context.Context, c client.Client
 
 	return nil
 }
+
+// ClusterWithoutImportedAnnotation returns a predicate that returns true only if the provided resource does not contain
+// "clusterImportedAnnotation" annotation. When annotation is present on the resource, controller will skip reconciliation.
+func ClusterWithoutImportedAnnotation(logger logr.Logger) predicate.Funcs {
```

Comment:
Shall we move this to a separate predicates package?

Reply:
this can be done in a follow up PR

Reply:
Done